src/software/ai/hl/stp/skill/dribble/dribble_skill_fsm.h

```diff
@@ -105,112 +91,121 @@ struct DribbleFSM
      *
      * This action will orient the robot towards the destination, dribble to the
      * destination, and then pivot to face the expected orientation
      *
-     * @param event DribbleFSM::Update
+     * @param event the Update event
      */
     void dribble(const Update &event);
 
     /**
-     * Start dribbling
-     *
-     * @param event DribbleFSM::Update
-     */
-    void startDribble(const Update &event);
-
-    /**
-     * Action to lose possession of the ball
+     * Action to lose BallControl of the ball
      *
-     * @param event DribbleFSM::Update
+     * @param event the Update event
      */
     void loseBall(const Update &event);
 
     /**
-     * Guard that checks if the robot has possession of the ball
+     * Guard that checks if the robot has control of the ball
      *
-     * @param event DribbleFSM::Update
+     * @param event the Update event
      *
-     * @return if the ball has been have_possession
+     * @return if the robot has control of the ball
      */
-    bool havePossession(const Update &event);
+    bool haveBallControl(const Update &event);
 
     /**
-     * Guard that checks if the robot has lost possession of the ball
+     * Guard that checks if the robot has lost control of the ball
      *
-     * @param event DribbleFSM::Update
+     * @param event the Update event
      *
-     * @return if the ball possession has been lost
+     * @return if the robot has lost control of the ball
      */
-    bool lostPossession(const Update &event);
+    bool lostBallControl(const Update &event);
 
     /**
      * Guard that checks if the ball is at the dribble_destination and robot is facing
-     * the right direction with possession of the ball
+     * the right direction with control of the ball
      *
-     * @param event DribbleFSM::Update
+     * @param event the Update event
      *
      * @return if the ball is at the dribble_destination, robot is facing the correct
-     * direction and ahs possession of the ball
+     * direction and has control of the ball
      */
     bool dribblingDone(const Update &event);
 
     /**
-     * Guard that checks if the the robot should lose possession to avoid excessive
-     * dribbling
+     * Guard that checks if the the robot should lose control of the ball to avoid
+     * excessive dribbling
      *
-     * @param event DribbleFSM::Update
+     * @param event the Update event
      *
-     * @return if the ball possession should be lost
+     * @return if the ball should be lost
      */
     bool shouldLoseBall(const Update &event);
 
+    /**
+     * Guard that checks whether the excessive_dribbling_mode control param
+     * is set to TbotsProto::ExcessiveDribblingMode::TERMINATE
+     *
+     * @param event the Update event
+     *
+     * @return whether the excessive_dribbling_mode control param is set
+     * to TbotsProto::ExcessiveDribblingMode::TERMINATE
+     */
+    bool terminateIfExcessiveDribbling(const Update &event);
```

Comment:
nit: `isExcessiveDribblingAllowed` is better. `terminate` implies that this is an action which it isn't

Reply:
or `shouldExcessivelyDribble` is consistent with the rest of this class